mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala

```diff
@@ -43,10 +43,13 @@ import org.apache.spark.sql.DataFrame
 @Since("1.0.0")
 class BinaryClassificationMetrics @Since("1.3.0") (
     @Since("1.3.0") val scoreAndLabels: RDD[(Double, Double)],
-    @Since("1.3.0") val numBins: Int) extends Logging {
+    @Since("1.3.0") var numBins: Int) extends Logging {
```

Comment:
You can't do it this way, because it won't cause the count/confusion RDDs to recompute. You can already set numBins right here right? what's the value in a setter?